OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -163,14 +163,33 @@ public void AddSmudge(CPos loc)
 			}
 		}
 
+		public void RemoveSmudge(CPos loc)
+		{
+			if (dirty.ContainsKey(loc))
+			{
+				var tile = dirty[loc];
+				tile.Depth = 0;
+				tile.Sprite = null;
+				dirty[loc] = tile;
+			}
+			else
+			{
+				var st = smudges.Keys.Random(world.SharedRandom);
+				dirty[loc] = new Smudge { Type = st, Depth = 0, Sprite = null };
+			}
+		}
+
 		public void TickRender(WorldRenderer wr, Actor self)
 		{
 			var remove = new List<CPos>();
 			foreach (var kv in dirty)
 			{
 				if (!self.World.FogObscures(kv.Key))
 				{
-					tiles[kv.Key] = kv.Value;
+					if (kv.Value.Sprite == null)
+						tiles.Remove(kv.Key);
+					else
+						tiles[kv.Key] = kv.Value;
```

Comment:
Isn't this redundant to the code commented above meaning that `render.Update` will already take care as it was dirtied?

Could this be the reason for #11660 because it changes game logic in a render tick and relies on `RenderPlayer` via `FogObscures`?